### PR TITLE
feat: add user profile view and components

### PR DESCRIPTION
**Pull request by owengretzinger**
### TL;DR
Added a new Profile component and updated the friends profile view with mock data

It shows a Remove Friend button if you're friends with them already, and an Add Friend button if not.

### What changed?
- Created a new Profile component with user information display and friend management
- Updated the friends profile view to use mock data and the new Profile component
- Modified the destructive button variant styling to use a softer, outlined appearance
- Replaced the basic ID display with a complete user profile interface

### How to test?
1. Navigate to a friend's profile page
2. Verify the profile displays mock user data (John Doe)
3. Confirm the profile picture, username, join date, and friend management buttons are visible
4. Check that the "Remove Friend" button shows the new outlined destructive styling
5. Verify the layout is centered and properly spaced

### CODE REVIEW

**Comment by owengretzinger** [comment]
> [!WARNING]
> <b>This pull request is not mergeable via GitHub because a downstack PR is open. Once all requirements are satisfied, merge this PR as a stack <a href="https://app.graphite.dev/github/pr/habits-together/app/133?utm_source=stack-comment-downstack-mergeability-warning" >on Graphite</a>.</b>
> <a href="https://graphite.dev/docs/merge-pull-requests">Learn more</a>

* **#133** <a href="https://app.graphite.dev/github/pr/habits-together/app/133?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈 <a href="https://app.graphite.dev/github/pr/habits-together/app/133?utm_source=stack-comment-view-in-graphite" target="_blank">(View in Graphite)</a>
* **#132** <a href="https://app.graphite.dev/github/pr/habits-together/app/132?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* `main`

<h2></h2>

This stack of pull requests is managed by <a href="https://graphite.dev?utm-source=stack-comment"><b>Graphite</b></a>. Learn more about <a href="https://stacking.dev/?utm_source=stack-comment">stacking</a>.

**Review by AnkushSarkar10 — APPROVED** [review]
